Tracker/Onboarding/PageViewController.swift

```diff
@@ -36,39 +42,53 @@ final class PageViewController: UIPageViewController {
     private func setupPageViewController() {
         dataSource = self
         delegate = self
-        if let first = pages.first {
-            setViewControllers([first], direction: .forward, animated: true)
-        }
+    
+        guard let firstViewController = pagesFactory?.getViewController(at: 0) else { return }
+        
+        setViewControllers(
+            [firstViewController],
+            direction: .forward,
+            animated: true
+        )
     }
     
     private func addViews() {
-        view.addSubview(pageControll)
+        view.addSubview(pageControl)
     }
     
     private func activateConstraints() {
-        let pageControllTopConstant = view.frame.height / 1.45
+        let pageControlTopConstant = view.frame.height / 1.45
         
         NSLayoutConstraint.activate([
-            pageControll.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: pageControllTopConstant),
-            pageControll.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+            pageControl.topAnchor.constraint(
+                equalTo: view.safeAreaLayoutGuide.topAnchor,
+                constant: pageControlTopConstant
+            ),
+            pageControl.centerXAnchor.constraint(equalTo: view.centerXAnchor)
         ])
     }
 }
 
 // MARK: UIPageViewControllerDataSource
 extension PageViewController: UIPageViewControllerDataSource {
-    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
-        guard let viewControllerIndex = pages.firstIndex(of: viewController) else { return nil }
-        let previousIndex = viewControllerIndex - 1
-        guard previousIndex >= 0 else { return nil }
-        return pages[previousIndex]
+    func pageViewController(
+        _ pageViewController: UIPageViewController,
+        viewControllerBefore viewController: UIViewController
+    ) -> UIViewController? {
+        guard let viewControllerIndex = pagesFactory?.getViewControllerIndex(of: viewController) else { return nil}
+        let prevIndex = viewControllerIndex - 1
+        guard prevIndex >= 0 else { return nil}
```

Comment:
заметила что у закрывающих скобок часто пробела нет
```suggestion
        guard prevIndex >= 0 else { return nil }
```